### PR TITLE
save metrics plot once the training is completed

### PR DESCRIPTION
**Pull request by SkalskiP**
# Description

Added a new class `MetricsPlotSink` that uses the existing callback system to store and save metric values. After training is complete, the loss values obtained at the end of each epoch for the train set and validation set are plotted, as well as the mAP and mAR values for the base and EMA models.

To unlock this feature, a new callback type `on_train_end` has been added.

example plot

### CODE REVIEW

**Review by isaacrob-roboflow — APPROVED** [review]
I'd be interested in ap50 too since we report those in the benchmark, but leaving that up to you! nice clean interface here :)
